### ISSUE TITLE
Markdoc: 'An argument for 'legacyConfig' was not provided.' in astro.config.ts

### ISSUE THREAD

**Issue by paulrudy** [withastro/astro#6993]
### What version of `astro` are you using?

2.4

### Are you using an SSR adapter? If so, which one?

Vercel

### What package manager are you using?

pnpm

### What operating system are you using?

Mac

### What browser are you using?

Safari, Chrome

### Describe the Bug

In VSCode, in `astro.config.ts`:
```ts
export default defineConfig({
  // ...
  integrations: [markdoc()]
  // ...
})
```
I get an error at `markdoc()`: 
```
Expected 1 arguments, but got 0.ts(2554)
index.d.ts(2, 44): An argument for 'legacyConfig' was not provided.
(alias) markdoc(legacyConfig: any): AstroIntegration
import markdoc
```

<img width="769" alt="Screenshot 2023-05-04 at 11 40 51 AM" src="https://user-images.githubusercontent.com/1110792/236304196-63cf15e8-9b5b-4525-ae01-e601088b0c19.png">

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-hfpu7v?file=astro.config.ts

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
@bholmesdev should this be optional?

**Comment by bholmesdev** [comment]
@matthewp Yes it should be! Easy fix

**Comment by bholmesdev** [comment]
Resolved by #7045
